src/ophyd_async/epics/pmac/_pmacTrajectory.py

```diff
@@ -0,0 +1,128 @@
+import time
+
+from bluesky.protocols import Flyable, Preparable
+
+from ophyd_async.core.async_status import AsyncStatus, WatchableAsyncStatus
+from ophyd_async.core.utils import WatcherUpdate
+from ophyd_async.epics.pmac import Pmac, PmacCSMotor
+
+TICK_S = 0.000001
+
+
+class PmacTrajectory(Pmac, Flyable, Preparable):
+    """Device that moves a PMAC Motor record"""
+
+    def __init__(
+        self, prefix: str, cs: int, motors: list[PmacCSMotor], name=""
+    ) -> None:
+        # Make a dict of which motors are for which cs axis
+        self.motors = {}
+        for motor in motors:
+            self.motors[motor.csAxis] = motor
+        self._fly_start: float
+        self.cs = cs
+        super().__init__(prefix, cs, name=name)
+
+    async def _ramp_up_velocity_pos(
+        self, velocity: float, motor: PmacCSMotor, end_velocity
+    ):
+        # Assuming ramping to or from 0
+        max_velocity_acceleration_time = await motor.acceleration_time.get_value()
+        max_velocity = await motor.max_velocity.get_value()
+        accl_time = max_velocity_acceleration_time * end_velocity / max_velocity
+        disp = 0.5 * (velocity + end_velocity) * accl_time
+        return [disp, accl_time]
+
+    @AsyncStatus.wrap
+    async def prepare(self, scanSpecStack):
```

Comment:
We can assume that the plan has given us a stack of frames where the axes are Motor objects
```suggestion
    async def prepare(self, stack: List[Frames[Motor]]):
```